### ISSUE TITLE
[Feature request] Wine

### ISSUE THREAD

**Issue by 0DeltaWaves** [ix0rai/bodaciousberries#57]
Very nice mod! I know you mentioned jam and smoothies as a possible planned feature, but I think wine would be fun/interesting as well. Barrels could be used to make the wine - maybe right clicking the barrel with a juice bottle would "deposit" the juice, and over time wine would be created, which could then be "harvested" by right clicking with an empty bottle.

**Comment by eristhea** [comment]
thank you, and that's an interesting idea!
are you suggesting that the behaviour of vanilla barrels be overridden when using juice on them, or adding an all new type of barrel block?

**Comment by 0DeltaWaves** [comment]
Oops, I guess I didn't think about that. Ideally it would be a new block I think?

**Comment by eristhea** [comment]
that's what I was thinking as well, I'd name it something like the "fermenting barrel," and use the vanilla barrel in the crafting recipe. also, I think this would be ideal if it had entirely no GUI, just deposit the juice, wait (maybe some sort of visual indicator towards progress?), and then remove

**Comment by 0DeltaWaves** [comment]
Yeah, I think that would work. I know Mr. Crayfish's Vehicles Mod uses small indicator bars/meters that appear when you mouse over barrels. In that mod it's for showing the amount of fluid in a barrel, but for this one it could be a progress bar?
Alternatively, a more immersive indicator could be a small particle effect coming from the barrel when the wine is done

**Comment by eristhea** [comment]
my idea was a bar on the texture itself, but yes I think that might be better. It's just a little more work as I would have to look into how he pulled it off.

**Comment by 0DeltaWaves** [comment]
I just edited my previous comment with an alternative idea (not sure if you saw it)  but I'll rewrite it here:
Alternatively, a more immersive indicator could be a small particle effect coming from the barrel when the wine is done

**Comment by eristhea** [comment]
hmm, how about both?
I can see myself mostly just settling a pile of barrels in a corner and waiting for the particle effect to show up, but it's always nice to have a progress bar

**Comment by 0DeltaWaves** [comment]
Yeah, that makes sense. It would be a great addition imo! Thanks for taking my idea into consideration.

**Comment by eristhea** [comment]
of course!
it'll probably be a pretty long time until I can find the time to build such a big feature (currently working on a brand new mod) but I'll be thinking about the ideal way to do this
